src/gr11review/part2/Utility.java

```diff
@@ -1,5 +1,393 @@
+
+    int[] intReturn;
+    int intCount;
+    int intLength;
+    int intSub;
+    int intLeft;
+    int intRight;
+
+    intLength = nums.length;
+    intReturn = new int[intLength];
+
+    for(intCount = 1; intCount < intLength - 1; intCount++){
+      intSub = nums[intCount];
+      if(intSub == value){
+        // check the numbers to its left and right
+        intLeft = nums[intCount - 1];
+        intRight = nums[intCount + 1];
+        // see which one is higher
+        if(intLeft < intRight){
+          // replace nums[intCount] with the higher number
+          intReturn[intCount] = intRight;
+        }else{
+          intReturn[intCount] = intLeft;
+        }
+        
+      }else{
+        intReturn[intCount] = nums[intCount];
+      }
+    }
+
+    return intReturn;
+  }
+  public static boolean canBalance(int[] nums) {
+
+    // Given a non-empty array, return true if there is a place to split the array so that the sum of the numbers on one side is equal to the sum of the numbers on the other side.
+    
+    int intCount;
+    int intCombined = 0;
+    int[] intArray = nums;
+    boolean returnTrue;
+    returnTrue = true;
+
+    for (intCount = 0; intArray != null; intCount++) {
+      intCombined = intCombined + nums[intCount];
+
+      if (intCombined % 2 != 0) {
+      returnTrue = false;
+      }else if (intCombined % 2 == 0) {
+      returnTrue = true;
+      }
+    }
+    return returnTrue;
+  }
+
+    public static void diagonal(int n) throws IOException {
+  // Given an integer n, write a method diagonal(int n) that outputs to a text file diagonalOut.txt, a two-dimensional array of size (n×n) populated as follows, with a comma between each number:
+
+  // The positions on the minor diagonal (from the upper right to the lower left corner) receive 1 .
+  // The positions above this diagonal receive 0 .
+  // The positions below the diagonal receive 2 .
+
+  int[][] intDiagonal;
+  int intCount;
+  int intCount2;
+  int intLine = 0;
+
+  intDiagonal = new int[n][n];
+
+// column number start from furthest right
+  for(intCount = n - 1; intCount >= 0; intCount --){
+    // Row number start from top
+    for(intCount2 = 0; intCount2 < n; intCount2 ++){
+      intDiagonal[intLine][intCount] = 1;
+      if(intCount2 > intLine){
+        intDiagonal[intCount2][intCount] = 2;
+      }else if(intCount2 < intLine){
+        intDiagonal[intCount2][intCount] = 0;
+      }
+    }
+    intLine = intLine + 1;
+  }
+
+  PrintWriter Output = new PrintWriter(new FileWriter("src/gr11review/part2/diagonalOut.txt"));
+
+  for(intCount2 = 0; intCount2 < n; intCount2++){
+    for(intCount = 0; intCount < n; intCount++){
+      Output.print(intDiagonal[intCount2][intCount] + ", ");
+    }
+    Output.println("");
+  }
+
+  Output.close();
+
+  }
+
+  public static String zipzap(String strInput){
```

Comment:
@AlexMeng21 
include javadoc commenting for all your methods.  Should also include author when working in team scencarios.